test/eni/IPRelease.go

```diff
@@ -0,0 +1,172 @@
+		cn := kubectl.GetCiliumNode(nodeName)
+		if cn == nil {
+			return false
+		}
+		return len(cn.Status.IPAM.ReleaseIPs) == 0
+	}
+}
+
+var _ = Describe("EniIPReleaseHandshakeTest", func() {
+	var (
+		kubectl           *helpers.Kubectl
+		ciliumCli         *helpers.CiliumCli
+		deploymentManager *helpers.DeploymentManager
+		releaseTestNs     = "cilium-integration-tests"
+	)
+
+	BeforeAll(func() {
+		var log = logging.DefaultLogger
+		var logger = logrus.NewEntry(log)
+		kubectl = helpers.CreateKubectl("", logger)
+		ciliumCli = helpers.CreateCiliumCli(logger)
+		deploymentManager = helpers.NewDeploymentManager()
+		deploymentManager.SetKubectl(kubectl)
+	})
+
+	AfterAll(func() {
+		deploymentManager.DeleteAll()
+	})
+
+	AfterFailed(func() {
+		kubectl.CiliumReport("cilium status", "cilium endpoint list")
+	})
+
+	JustAfterEach(func() {
+		kubectl.ValidateNoErrorsInLogs(CurrentGinkgoTestDescription().Duration)
+	})
+
+	It("Test IP release abort", func() {
+		By("Enabling aws excess IP release feature and restarting cilium operator")
+		err := ciliumCli.UpdateCiliumConfig("aws-release-excess-ips", "true", false)
+		Expect(err).NotTo(HaveOccurred())
+		err = ciliumCli.UpdateCiliumConfig("excess-ip-release-delay", "30", false)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Workaround till cilium-cli supports restarting operator based on config change
+		err = kubectl.RestartOperator()
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Create nodegroup to deploy tests on")
+		ng := nodegroups.EksNodegroup{NodePool: nodegroups.NodePool{
+			Name:          "ip-release",
+			ClusterName:   os.Getenv("clusterName"),
+			InstanceTypes: []string{"t3.medium", "t3a.medium"},
+			Region:        os.Getenv("region"),
+			NumNodes:      1,
+			NodeTaint:     "ip-release",
+			Executor:      helpers.CreateLocalExecutor(os.Environ()),
+			Kubectl:       kubectl,
+		}}
+		err = ng.CreateNodePool()
+		Expect(err).NotTo(HaveOccurred(), "Error while creating nodegroup")
+
+		By("Creating a deployment to pick a node to test on")
+		configFile := "nginx-with-tolerations.yaml"
+		labelSelector := "zgroup=http-server"
+		tmpl := helpers.ManifestGet(kubectl.BasePath(), "http-deployment-node-toleration.yaml.tmpl")
+		helpers.RenderTemplateWithData(tmpl, configFile, struct{ NodeTaint string }{NodeTaint: "ip-release"})
+
+		_ = kubectl.Exec(fmt.Sprintf("kubectl create ns %s", releaseTestNs))
+		res := kubectl.Exec(fmt.Sprintf("kubectl -n %s apply -f %s", releaseTestNs, configFile))
+		res.ExpectSuccess("Error creating test workload: %s", res.OutputPrettyPrint())
+		kubectl.WaitforNPodsRunning(releaseTestNs, "-l "+labelSelector, 2, 10*time.Minute)
+
+		By("Fetching cilium node for test pods")
+		data, err := kubectl.GetPodsNodes(releaseTestNs, labelSelector)
+		Expect(err).To(BeNil(), "error getting pod->node mapping")
+		var nodeName string
+		for _, v := range data {
+			nodeName = v
+			break
+		}
+
+		cn := kubectl.GetCiliumNode(nodeName)
+		Expect(cn).NotTo(Equal(check.IsNil))
```

Comment:
```suggestion
		Expect(cn).NotTo(BeNil())
```

If you're using Ginkgo, you don't need `check` pkg because Ginkgo provides matchers to do these things